AppDevPods/AppDevUIKit/ADKMultiGradientView.m

```diff
@@ -0,0 +1,92 @@
+//
+//  ADKMultiGradientView.m
+//  AppDevKit
+//
+//  Created by  Chih Feng Sung on 1/29/19.
+//  Copyright © 2019 Yahoo. All rights reserved.
+//
+
+#import "ADKMultiGradientView.h"
+
+@interface ADKMultiGradientView ()
+
+//@property (nonatomic, strong) NSMutableArray *colors;
+//@property (nonatomic, strong) CGFloat *locations;
+
+@end
+
+@implementation ADKMultiGradientView
+
+- (instancetype)initWithCoder:(NSCoder *)coder
+{
+    if (self = [super initWithCoder:coder]) {
+        [self setupView];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    if (self = [super initWithFrame:frame]) {
+        [self setupView];
+    }
+    return self;
+}
+
+- (void)setupView
+{
+    self.blendsType = ADKBlendsTypeFromTopToBottom;
+}
+
+-(void)redrawView
+{
+    [self setNeedsDisplay];
+}
+
+- (void)drawRect:(CGRect)rect
+{
+    [super drawRect:rect];
+
+    CGContextRef contextRef = UIGraphicsGetCurrentContext();
+    CGColorSpaceRef colorSpaceRef = CGColorSpaceCreateDeviceRGB();
+
+    NSMutableArray *colors = [NSMutableArray arrayWithCapacity:self.gradientColors.count];
+    for (int i = 0; i < self.gradientLocations.count; i++) {
+        colors[i] =(id)[(UIColor *)[self.gradientColors objectAtIndex:i] CGColor];
+    }
+
+    CGFloat locations[4];
```

Comment:
Might have issue if color not equal 4
```suggestion
    CGFloat locations[self.gradientLocations.count];
```

Reply:
I am speechless... I must be too tired to write this. >///<